pandas/core/arrays/integer.py

```diff
@@ -206,7 +206,17 @@ def coerce_to_array(values, dtype, mask=None, copy=False):
             mask = mask.copy()
         return values, mask
 
-    values = np.array(values, copy=copy)
+    if hasattr(values, "dtype"):
+        if isinstance(values, Categorical):
+            values = np.array(values, copy=copy)
+        elif copy:
+            values = values.copy()
+    else:
+        if isinstance(values, list):
+            values = np.array(values, dtype=object, copy=copy)
+        else:
+            values = np.array(values, copy=copy)
```

Comment:
Is there a reason that this if/else is needed? Can't we do the `np.array(values, dtype=object, copy=copy)` in all cases?

Reply:
They both work equally. And I have updated that.
But I am not understanding why [try_cast_to_ea(self._values, new_values)](https://github.com/pandas-dev/pandas/blob/e6a3993f0cfaa75d3bd7c43fe1e7f3bff7dd4c3c/pandas/core/series.py#L2591l)
returns int values for `new_values=list of bools` and `self._values=Series of integers`.
[Find test log here](https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=23710&view=logs&j=bef1c175-2c1b-51ae-044a-2437c76fc339&t=9365e6ae-bf62-55c3-8ba1-d11b71b622f9&l=113).

Reply:
>But I am not understanding why try_cast_to_ea(self._values, new_values)
returns int values for new_values=list of bools and self._values=Series of integers.

I think it creates an Extension array of type of values in self._value. Can we pass something else there as the first argument in order to get the EA of the type of new_values?

Reply:
Okay, the only problem is that whenever it gets self._values  as integer EA and the new_values is a list of boolean then it returns EA of ints instead of EA of bools. Can we just put a condition for that case or is there any other suggestion @jorisvandenbossche

Reply:
>Can we just put a condition for that case or is there any other suggestion @jorisvandenbossche

I tried that and that works fine. There is only 1 failing test now which is that the combine function is returning bool Series instead of Sparse Series. 
**In which case, combine function should return Sparse Series?** Particularly in the case of the argument `func` being a lambda function, or is it something else?